kube-controllers/pkg/controllers/flannelmigration/ipam_migrator.go

```diff
@@ -363,11 +363,19 @@ func createDefaultVxlanIPPool(ctx context.Context, client client.Interface, cidr
 // If migrating from Flannel, return error if vxlan is not enabled.
 // If migrating from Canal, set vxlan enabled.
 // Do nothing if correct values already been set.
-func updateDefaultFelixConfigurtion(ctx context.Context, client client.Interface, vni, port, mtu int, checkVxlan bool) error {
+func updateOrCreateDefaultFelixConfiguration(ctx context.Context, client client.Interface, vni, port, mtu int, checkVxlan bool) error {
 	// Get default Felix configuration. Return error if not exists.
 	defaultConfig, err := client.FelixConfigurations().Get(ctx, defaultFelixConfigurationName, options.GetOptions{})
+	if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+		// Create the default config if it doesn't already exist.
+		defaultConfig = api.NewFelixConfiguration()
+		defaultConfig.Name = defaultFelixConfigurationName
+		t := true
+		defaultConfig.Spec.VXLANEnabled = &t
+		defaultConfig, err = client.FelixConfigurations().Create(ctx, defaultConfig, options.SetOptions{})
+	}
 	if err != nil {
-		log.WithError(err).Errorf("Error getting default FelixConfiguration resource")
+		log.WithError(err).Errorf("Error creating default FelixConfiguration resource")
```

Comment:
This isn't right - we can enter this branch either after a failed create or after a failed get (if the failure wasn't a ResourceNotExist error) 

We should handle each of those error cases individually rather than letting them fall through to the same error handling branch.

Reply:
Will fix. I also think this will change once I do away with needing to look into the FelixConfig VXLAN setting here...